homeassistant/components/luxor/light.py

```diff
@@ -0,0 +1,199 @@
+"""Light support for FX Luminaire Luxor low voltage controller integration."""
+
+from datetime import timedelta
+import logging
+
+from aiohttp import client_exceptions as aio_exceptions
+import async_timeout
+from luxor import LuxorError
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    SUPPORT_BRIGHTNESS,
+    LightEntity,
+)
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import (
+    CONF_INCLUDE_LUXOR_THEMES,
+    DEFAULT_INCLUDE_LUXOR_THEMES,
+    DOMAIN as LUXOR_DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Luxor light platform.
+
+    Adds groups and themes from the Luxor controller as light entities.
+    """
+    client = hass.data[LUXOR_DOMAIN][config_entry.entry_id]
+
+    async def async_update_group_data():
+        """Fetch group data from Luxor controller."""
+        groups = await async_safe_fetch(client.get_groups)
+        return {group["Grp"]: group for group in groups}
+
+    light_coordinator = DataUpdateCoordinator(
```

Comment:
should we setup the coordinators in the integration init? That would allows easier addition of other platforms in the future.

Additionally, the first refresh of the coordinator can throw exceptions the integration setup can use.